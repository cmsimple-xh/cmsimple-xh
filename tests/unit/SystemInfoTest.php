<?php

/**
 * Testing the mailform.
 *
 * @author    The CMSimple_XH developers <devs@cmsimple-xh.org>
 * @copyright 2025 The CMSimple_XH developers <http://cmsimple-xh.org/?The_Team>
 * @license   http://www.gnu.org/licenses/gpl-3.0.en.html GNU GPLv3
 * @see       http://cmsimple-xh.org/
 */

namespace XH;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

class SystemInfoTest extends TestCase
{
    protected function setUp(): void
    {
        global $pth, $cf, $tx;

        $pth = [
            'folder' => [
                'cmsimple' => '',
                'content' => '',
                'corestyle' => '',
                'images' => '',
                'downloads' => '',
                'userfiles' => '',
                'media' => '',
            ],
            'file' => [
                'config' => '',
                'log' => '',
                'debug-log' => '',
                'language' => '',
                'content' => '',
                'template' => '',
                'stylesheet' => ''
            ],
        ];
        $cf = XH_includeVar('./cmsimple/config.php', 'cf');
        $tx = XH_includeVar('./cmsimple/languages/en.php', 'tx');
    }

    public function testRendersSystemInfo(): void
    {
        $systemInfo = $this->getMockBuilder(SystemInfo::class)->onlyMethods([
            'plugins', 'isAccessProtected', 'isExtensionLoaded', 'getIni', 'defaultTimezone', 'setLocale',
        ])->getMock();
        $systemInfo->expects($this->any())->method('plugins')->willReturn(['filebrowser']);
        $systemInfo->expects($this->any())->method('isAccessProtected')->willReturn(true);
        $systemInfo->expects($this->any())->method('isExtensionLoaded')->willReturn(true);
        $systemInfo->expects($this->any())->method('getIni')->willReturnMap([
            ['safe_mode', ''],
            ['session.use_trans_sid', ''],
            ['session.use_only_cookies', '1'],
            ['session.cookie_lifetime', ''],
        ]);
        $systemInfo->expects($this->any())->method('defaultTimezone')->willReturn('Europe/Berlin');
        $systemInfo->expects($this->any())->method('setLocale')->willReturnArgument(1);
        Approvals::verifyHtml($systemInfo->render());
    }
}
