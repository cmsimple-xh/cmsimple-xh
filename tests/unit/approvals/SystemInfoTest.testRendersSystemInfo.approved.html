<h2>Installed CMSimple Version</h2>
<ul>
<li>CMSimple_XH 1.8&nbsp;&nbsp;Released: 2024-12-11</li>
</ul>
<h2>Installed Plugins</h2>
<ul>
<li>Filebrowser for CMSimple_XH 1.8</li>
</ul>
<h2>Webserver</h2>
<ul>
<li>Webserver could not be determined</li>
</ul>
<h2>PHP-Version</h2>
<ul>
<li>7.4.19</li>
<li><a href="?&phpinfo" target="_blank"><b>PHP Info &raquo;</b></a> &nbsp; (opens in a new window or tab)</li>
</ul>
<h2>Info and Help Links</h2>
<ul>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://www.cmsimple-xh.org/">cmsimple-xh.org &raquo;</a></li>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://wiki.cmsimple-xh.org/">wiki.cmsimple-xh.org &raquo;</a></li>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://www.cmsimpleforum.com/">cmsimpleforum.com &raquo;</a></li>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://www.cmsimple-xh.org/?Important-Links">cmsimple-xh.org/?Important-Links &raquo;</a></li>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://www.cmsimple-xh.org/?Plugin-Repository">cmsimple-xh.org/?Plugin-Repository &raquo;</a></li>
<li><a target="_blank" rel="noopener" rel="noreferrer" href="https://www.cmsimple-xh.org/?Template-Repository">cmsimple-xh.org/?Template-Repository &raquo;</a></li>
</ul>
<h2>System check</h2>
<ul id="xh_system_check">
<li class="xh_success ">Checking that PHP version ≥ 7.4.0 … okay</li>
<li class="xh_success xh_system_check_cat_start">Checking that extension 'intl' is loaded … okay</li>
<li class="xh_success ">Checking that extension 'json' is loaded … okay</li>
<li class="xh_success ">Checking that extension 'mbstring' is loaded … okay</li>
<li class="xh_success ">Checking that extension 'openssl' is loaded … okay</li>
<li class="xh_success ">Checking that extension 'session' is loaded … okay</li>
<li class="xh_success ">Checking that extension 'curl' is loaded … okay</li>
<li class="xh_success xh_system_check_cat_start">Checking that function 'fsockopen' is available … okay</li>
<li class="xh_warning xh_system_check_cat_start">Checking that '' is writable … warning</li>
<li class="xh_warning ">Checking that '.sessionname' is writable … warning</li>
<li class="xh_success xh_system_check_cat_start">Checking that <a target="_blank" href="">'' is access protected</a> … okay</li>
<li class="xh_success ">Checking that <a target="_blank" href="">'' is access protected</a> … okay</li>
<li class="xh_success ">Checking that <a target="_blank" href="">'' is access protected</a> … okay</li>
<li class="xh_success ">Checking that <a target="_blank" href="">'' is access protected</a> … okay</li>
<li class="xh_success ">Checking that <a target="_blank" href="">'' is access protected</a> … okay</li>
<li class="xh_success ">Checking that default locale is in use … okay</li>
<li class="xh_success ">Checking that time zone is valid … okay</li>
<li class="xh_success ">Checking that safe_mode is off … okay</li>
<li class="xh_success ">Checking that session.use_trans_sid is off … okay</li>
<li class="xh_success ">Checking that session.use_only_cookies is on … okay</li>
<li class="xh_success ">Checking that session.cookie_lifetime is 0 … okay</li>
<li class="xh_success ">Checking that there is no <a href="https://wiki.cmsimple-xh.org/?glossar#id7_bom" target="_blank">BOM</a> … okay</li>
<li class="xh_warning ">Checking that non-default password is set … warning</li>
</ul>
